scripts/cost-calculator.py

```diff
@@ -25,18 +25,35 @@
 total_days = (diff +1)
 weekend_days = (total_days - business_days)
 
-def azPriceAPI(vm_sku, productNameVar, osQuery):
-    #Microsoft Retail Rates Prices API query and response. (https://learn.microsoft.com/en-us/rest/api/cost-management/retail-prices/azure-retail-prices)
-    api_url = "https://prices.azure.com/api/retail/prices?currencyCode='GBP&api-version=2021-10-01-preview"
-    query = "armRegionName eq 'uksouth' and skuName eq '" + vm_sku + "' and priceType eq 'Consumption' and productName eq " + osQuery
-    response = requests.get(api_url, params={'$filter': query})
-    json_data = json.loads(response.text)
+#FUnction to add entries to the GitHub env list.
+def writeVar(varName, varValue):
+    with open(env_file_path, 'a') as env_file:
+        env_file.write('\n' + varName + "=" + str(varValue) + '\n')
+        env_file.close()
 
-    #Get retail price from json API response
-    for item in json_data['Items']:
-        vm_hour_rate = item['retailPrice']
-    
-    return vm_hour_rate
+def azPriceAPI(vm_sku, productNameVar, osQuery,retry=0):
+    try:
+        #Microsoft Retail Rates Prices API query and response. (https://learn.microsoft.com/en-us/rest/api/cost-management/retail-prices/azure-retail-prices)
+        api_url = "https://prices.azure.com/api/retail/prices?currencyCode='GBP&api-version=2021-10-01-preview"
+        query = "armRegionName eq 'uksouth' and skuName eq '" + vm_sku + "' and priceType eq 'Consumption' and productName eq " + osQuery
+        response = requests.get(api_url, params={'$filter': query})
+        json_data = json.loads(response.text)
+        #Get retail price from json API response
+        for item in json_data['Items']:
+            vm_hour_rate = item['retailPrice']
+        
+        return vm_hour_rate
+
+    #API occasionally fails to return a value which was causing issues in cost feedback to users. See DTSPO-15193
+    #Retry will attempt up to 50 retries. If it is still unable to return a value, the rate will be defaulted to 0.
+    except:
+        if retry < 50: #Edit retry limit here.
```

Comment:
I wouldn't retry that hard, the person won't see the cost anyway if it's not done quickly
```suggestion
        if retry < 5: #Edit retry limit here.
```